src/subdomains/lightning/services/lightning-transaction.service.ts

```diff
@@ -0,0 +1,278 @@
+import {
+  TransactionLightningEntity,
+  TransactionLightningState,
+  TransactionLightningType,
+} from '../entities/transaction-lightning.entity';
+import { TransactionOnchainEntity } from '../entities/transaction-onchain.entity';
+import { UserTransactionEntity, UserTransactionType } from '../entities/user-transaction.entity';
+import { TransactionLightningRepository } from '../repositories/transaction-lightning.repository';
+import { TransactionOnchainRepository } from '../repositories/transaction-onchain.repository';
+import { UserTransactionRepository } from '../repositories/user-transaction.repository';
+
+@Injectable()
+export class LightningTransactionService {
+  private readonly client: LightningClient;
+
+  constructor(
+    lightningService: LightningService,
+    private readonly transactionOnchainRepo: TransactionOnchainRepository,
+    private readonly transactionLightningRepo: TransactionLightningRepository,
+    private readonly userTransactionRepository: UserTransactionRepository,
+    private readonly lightingWalletRepository: LightingWalletRepository,
+  ) {
+    this.client = lightningService.getDefaultClient();
+  }
+
+  async syncOnchainTransactions(): Promise<TransactionOnchainEntity[]> {
+    const maxId = await this.transactionOnchainRepo.getMaxId();
+
+    if (!maxId) {
+      return this.saveOnchainTransactions(0);
+    }
+
+    const transactionEntity = await this.transactionOnchainRepo.findOneBy({ id: maxId });
+    const maxBlock = transactionEntity?.block ?? -1;
+
+    return this.saveOnchainTransactions(maxBlock + 1);
+  }
+
+  private async saveOnchainTransactions(startBlockHeight: number): Promise<TransactionOnchainEntity[]> {
+    const newTransactionEntities: TransactionOnchainEntity[] = [];
+
+    const onchainTransactions = (await this.client.getOnchainTransactions(startBlockHeight)).reverse();
+
+    for (const onchainTransaction of onchainTransactions) {
+      const transactionEntity = this.transactionOnchainRepo.create({
+        transaction: onchainTransaction.tx_hash,
+        amount: Number(onchainTransaction.amount),
+        fee: Number(onchainTransaction.total_fees),
+        block: onchainTransaction.block_height,
+        timestamp: new Date(Number(onchainTransaction.time_stamp) * 1000),
+      });
+
+      newTransactionEntities.push(transactionEntity);
+    }
+
+    if (newTransactionEntities.length > 0) {
+      newTransactionEntities[newTransactionEntities.length - 1].balance =
+        await this.client.getLndConfirmedWalletBalance();
+    }
+
+    return this.transactionOnchainRepo.saveMany(newTransactionEntities);
+  }
+
+  async syncLightningTransactions(): Promise<TransactionLightningEntity[]> {
+    const maxId = await this.transactionLightningRepo.getMaxId();
+
+    if (!maxId) {
+      return this.saveLightningTransactions(null);
+    }
+
+    const databaseTransactionEntity = await this.transactionLightningRepo.findOneBy({ id: maxId });

```

Comment:
Same here

Reply:
Other implementation, not needed anymore.